gnmi/gnmi.go

```diff
@@ -215,10 +225,10 @@ func updateCacheNotifs(cache *cache.Cache, nos []*gpb.Notification, target, orig
 //
 // update indicates whether to update the cache with the values from the set
 // request.
-func set(schema *ytypes.Schema, cache *cache.Cache, target string, req *gpb.SetRequest, preferShadowPath bool) error {
+func set(schema *ytypes.Schema, cache *cache.Cache, target string, req *gpb.SetRequest, preferShadowPath bool, validators []func(*oc.Root) error) error {
 	prevRoot, err := ygot.DeepCopy(schema.Root)
 	if err != nil {
-		return fmt.Errorf("gnmi: failed to ygot.DeepCopy the cached root object: %v", err)
+		return status.Errorf(codes.Internal, "failed to ygot.DeepCopy the cached root object: %v", err)
```

Comment:
This is an internal function, should it rather return a normal error, and leave the task of wrapping it in a gRPC status to the server implementation? (i.e. Set in this case)

Reply:
this returns different codes so we can't wrap it universally

Reply:
OK, can you add a comment on this behaviour for set?